core/ui/ui.go

```diff
@@ -1,13 +1,41 @@
 package ui
 
+import "errors"
+
 // Config is the UI configuration for core.Endpoint
 type Config struct {
 	// HideHostname whether to hide the hostname in the Result
 	HideHostname bool `yaml:"hide-hostname"`
 	// HideURL whether to ensure the URL is not displayed in the results. Useful if the URL contains a token.
 	HideURL bool `yaml:"hide-url"`
 	// DontResolveFailedConditions whether to resolve failed conditions in the Result for display in the UI
-	DontResolveFailedConditions bool `yaml:"dont-resolve-failed-conditions"`
+	DontResolveFailedConditions bool   `yaml:"dont-resolve-failed-conditions"`
+	Badge                       *Badge `yaml:"badge"`
+}
+
+type Badge struct {
+	ResponseTime []int `yaml:"response-time"`
+}
+
+var (
+	ErrInvalidBadgeResponseTimeConfig = errors.New("invalid response time badge configuration: expected parameter 'response-time' to have 5 ascending numerical values")
+)
+
+func (config *Config) ValidateAndSetDefaults() error {
+	if config.Badge != nil {
+		if len(config.Badge.ResponseTime) != 5 {
+			return ErrInvalidBadgeResponseTimeConfig
+		}
+		for i := 4; i > 0; i-- {
+			if config.Badge.ResponseTime[i] < config.Badge.ResponseTime[i-1] {
+				return ErrInvalidBadgeResponseTimeConfig
+			}
+		}
+		config.Badge.ResponseTime = GetDefaultConfig().Badge.ResponseTime
+	}
+
+	config.Badge = GetDefaultConfig().Badge
+	return nil
```

Comment:
I don't think this is right.

So we're checking if the badge is set, if it's set, then we validate it. That's good so far, but at line 34, we overwrite it with the default configuration's `Badge.ResponseTime`..?

And _after_ that, we overwrite it a second time at line 37 with the default config's `Badge`?

Maybe it's just because I'm mentally exhausted right now, but I don't think it makes sense 🤔

Reply:
I'm not into all details of the project right know. I will review this code.